src/reviews/views.py

```diff
@@ -327,3 +332,77 @@ def get_success_url(self):
         if query_string:
             return url + '?' + query_string
         return url
+
+
+def review_stages(request):
+    current_review_stages_setting = {}
+    review_stages_list = [
+        'Call for Proposals',
+        'Locked (proposal editing and reviewing disabled)',
+        'First Round Review', 'Modification Stage', 'Second Round Review',
+        'Internal Decision', 'Announcement of Acceptance'
+    ]
+    review_stages_var = [
+        'proposals.creatable', 'proposals.editable', 'proposals.withdrawable',
+        'reviews.visible.to.submitters', 'reviews.stage',
+        'proposals.disable.after'
+    ]
+
+    # Initialize current setting with existing value
+    for tag in review_stages_var:
+        key = settings.CONFERENCE_DEFAULT_SLUG + '.' + tag
+        value = reg.get(key, '')
+        update_current_review_stages_setting(tag, value, current_review_stages_setting)
+
+    if request.method == 'POST':
+        for tag in review_stages_var:
+            key = settings.CONFERENCE_DEFAULT_SLUG + '.' + tag
+            if tag == 'proposals.disable.after':
+                if request.POST[tag] == "":
+                    continue
+                else:
+                    date_time_obj = date_preprocess(DATETIME_INPUT_FORMATS, request.POST[tag])
+                    if date_time_obj is None:
+                        messages.error(request, 'Please input valid date format : " + "%Y-%m-%dT%H:%M')
+                        value = None
+                    else:
+                        tz_selectd = pytz.timezone(request.POST['review_timezone'])
+                        loc_dt = tz_selectd.localize(date_time_obj).strftime(
+                            '%Y-%m-%d %H:%M:%S%z')
+                        value = loc_dt
+            elif tag == 'reviews.stage':
+                value = int(request.POST[tag])
+            else:
+                value = request.POST[tag]
+            reg[key] = value
+            update_current_review_stages_setting(tag, value, current_review_stages_setting)
+
+        messages.info(request, 'This setting has been changed successfully.')
+
+    return render(
+        request, 'reviews/review_stages.html', {
+            'timezones': pytz.common_timezones,
+            'review_stages_list': review_stages_list,
+            'current_review_stages_setting': current_review_stages_setting,
+            **reviews_state()._asdict()
+        })
+
+
+def update_current_review_stages_setting(tag, value, current_review_stages_setting):
+    # Django template language does not support dictionary keys containing "."
+    if "." in tag:
+        tag = tag.replace(".", "_")
+    current_review_stages_setting[tag] = value
```

Comment:
to relieve the maintainers/developers from the burden to remember the conversion logic, can we align the naming of the reg keys and context pass to the template?